### DISCUSSION TITLE
What the heck is up with the code?

### DISCUSSION THREAD

**Discussion by rcmaehl** [ThioJoe/YT-Spammer-Purge#181] [Q&A]
<redacted\>

I was just trying to add in functionality to have a blacklist definitions file for other languages for #176 but I'm over here having to run it through several different conversions for, what should honestly be, plain text for who knows why. 

Geez

Thanks in advance

**Comment by rbeer** [comment]
b0f45b9602b5d1a0ea0865ece25c0201c8554754

You just called yourself "the dumbest scammer" :rofl: 

[git-blame](https://www.git-scm.com/docs/git-blame) is your friend.

**Reply by rcmaehl** [reply]
> <redacted\>
> 
> You just called yourself "the dumbest scammer" 🤣

Not sure how. I successfully reverse engineered and got the original text, was just wondering *why*. The script is open source and the people who actually coded the bots can easily do the same in the same ~3 minutes it took me to realize what was going on. 

If trying to obfuscate then that function and it's variables should honestly be obfuscated as well so scammers have to follow the code path instead of CTRL-F

> [git-blame](https://www.git-scm.com/docs/git-blame) is your friend.

Good point. My fault for using github desktop 😅

**Reply by rbeer** [reply]
> Not sure how. 

That's what he wrote in the commit message and you fell for it; hence... But never mind; just playing with you :stuck_out_tongue: 

> Good point. My fault for using github desktop sweat_smile

Dayum; that's not a thing in Desktop? Issue is from 2017 desktop/desktop#2310
Anyway, it's in the webUI. Click a line number and then in the `···` menu.

**Comment by ThioJoe** [comment]
I know it's a trivially low barrier but It's basically just to stop any open-mouthed dribbling scammers from glancing over the code and and easily spotting the filters.

**Reply by rcmaehl** [reply]
It's definitely a barrier but honestly most of the dumber scammers can't do much in the first place. Scammers generally buy scripts, sites, numbers and other resources from a few "distributors" and only change the most basic things like the phone number. The actual distributors are going to be much smarter.

I'd really recommend getting more obfuscation in when you can.

Instructional strings can be converted to variables and stored elsewhere (also adds translation support)
Functions can be renamed to be more unassuming (e.g. check_video_data_smart)
Same with variables (e.g redadwords -> highdata, yellowAdWords -> mediumdata)
Split changes like you've already done across additional .py files imported as modules

I've redacted my original statements since the code was intentional, and you can delete those discussion revisions as repo owner. (@rbeer  can you redact as well?)

Just honestly be prepared for scammers to adapt once this project goes viral, and I'm sure it will

**Reply by ThioJoe** [reply]
I expect them to adapt. The goal is to require them to have to make such convoluted workarounds / names, etc, that they will become more obvious as scams to users. Hopefully to the point where it is not profitable.

The ultimate solution must come from Google/YouTube. This is just a stop gap

**Reply by domis86** [reply]
> The goal is to require them to have to make such convoluted workarounds / names, etc, that they will become more obvious as scams to users. Hopefully to the point where it is not profitable.

Thats good plan IMO :+1:
